temporalcli/commands.operator_search_attribute.go

```diff
@@ -0,0 +1,167 @@
+		Namespace: c.Parent.Parent.Namespace,
+	}
+	existingSearchAttributes, err := cl.OperatorService().ListSearchAttributes(cctx, listReq)
+	if err != nil {
+		return fmt.Errorf("unable to get existing search attributes: %w", err)
+	}
+
+	searchAttributes := make(map[string]enums.IndexedValueType, len(c.Type))
+	for i, saType := range c.Type {
+		saName := c.Name[i]
+		typeInt, err := stringToEnum(saType)
+		if err != nil {
+			return fmt.Errorf("unable to parse search attribute type %s: %w", saType, err)
+		}
+		existingSearchAttributeType, searchAttributeExists := existingSearchAttributes.CustomAttributes[saName]
+		if searchAttributeExists && existingSearchAttributeType != typeInt {
+			return fmt.Errorf("search attribute %s already exists and has different type %s", saName, existingSearchAttributeType.String())
+		}
+		searchAttributes[saName] = typeInt
+	}
+
+	request := &operatorservice.AddSearchAttributesRequest{
+		SearchAttributes: searchAttributes,
+		Namespace:        c.Parent.Parent.Namespace,
+	}
+
+	_, err = cl.OperatorService().AddSearchAttributes(cctx, request)
+	if err != nil {
+		return fmt.Errorf("unable to add search attributes: %w", err)
+	}
+	cctx.Printer.Println(color.GreenString("Search attributes have been added"))
+	return nil
+}
+
+func stringToEnum(search string) (enums.IndexedValueType, error) {
+	switch strings.ToLower(search) {
+	case "text":
+		return enums.INDEXED_VALUE_TYPE_TEXT, nil
+	case "keyword":
+		return enums.INDEXED_VALUE_TYPE_KEYWORD, nil
+	case "int":
+		return enums.INDEXED_VALUE_TYPE_INT, nil
+	case "double":
+		return enums.INDEXED_VALUE_TYPE_DOUBLE, nil
+	case "bool":
+		return enums.INDEXED_VALUE_TYPE_BOOL, nil
+	case "datetime":
+		return enums.INDEXED_VALUE_TYPE_DATETIME, nil
+	case "keywordlist":
+		return enums.INDEXED_VALUE_TYPE_KEYWORD_LIST, nil
+	default:
+		return enums.INDEXED_VALUE_TYPE_UNSPECIFIED, fmt.Errorf("unsupported search attribute type: %v, supported types are: [Text, Keyword, Int, Double, Bool, Datetime, KeywordList]", search)
+	}
+}
+
+func (c *TemporalOperatorSearchAttributeRemoveCommand) run(cctx *CommandContext, args []string) error {
+	cl, err := c.Parent.Parent.ClientOptions.dialClient(cctx)
+	if err != nil {
+		return err
+	}
+	defer cl.Close()
+
+	yes, err := cctx.promptYes(
+		fmt.Sprintf("You are about to remove search attributes %v? y/N", c.Name), c.Yes)
+	if err != nil {
+		return err
+	} else if !yes {
+		// We consider this a command failure
+		return fmt.Errorf("user denied confirmation, operation aborted")
+	}
+
+	names := c.Name
+	namespace := c.Parent.Parent.Namespace
+	if err != nil {
+		return err
+	}
+
+	request := &operatorservice.RemoveSearchAttributesRequest{
+		SearchAttributes: names,
+		Namespace:        namespace,
+	}
+
+	_, err = cl.OperatorService().RemoveSearchAttributes(cctx, request)
+	if err != nil {
+		return fmt.Errorf("unable to remove search attributes: %w", err)
+	}
+
+	// response contains nothing
```

Comment:
Would be ok w/ a simple println telling users it was removed similar to one we have for when one is added, but up to you.